FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJob.cs

```diff
@@ -0,0 +1,580 @@
+        /// The format is '{SchemaType}_{index:d10}.parquet', e.g. Patient_0000000001.parquet.
+        /// </summary>
+        private Dictionary<string, int> _outputFileIndexMap;
+
+        public FhirToDataLakeProcessingJob(
+            long jobId,
+            FhirToDataLakeProcessingJobInputData inputData,
+            IFhirDataClient dataClient,
+            IFhirDataWriter dataWriter,
+            IColumnDataProcessor parquetDataProcessor,
+            IFhirSchemaManager<FhirParquetSchemaNode> fhirSchemaManager,
+            ITypeFilterParser typeFilterParser,
+            IGroupMemberExtractor groupMemberExtractor,
+            FilterConfiguration filterConfiguration,
+            ILogger<FhirToDataLakeProcessingJob> logger)
+        {
+            _jobId = jobId;
+            _inputData = EnsureArg.IsNotNull(inputData, nameof(inputData));
+
+            _dataClient = EnsureArg.IsNotNull(dataClient, nameof(dataClient));
+            _dataWriter = EnsureArg.IsNotNull(dataWriter, nameof(dataWriter));
+            _parquetDataProcessor = EnsureArg.IsNotNull(parquetDataProcessor, nameof(parquetDataProcessor));
+            _fhirSchemaManager = EnsureArg.IsNotNull(fhirSchemaManager, nameof(fhirSchemaManager));
+            _typeFilterParser = EnsureArg.IsNotNull(typeFilterParser, nameof(typeFilterParser));
+            _groupMemberExtractor = EnsureArg.IsNotNull(groupMemberExtractor, nameof(groupMemberExtractor));
+            _filterConfiguration = EnsureArg.IsNotNull(filterConfiguration, nameof(filterConfiguration));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        // the processing job status is never set to failed or cancelled.
+        public async Task<string> ExecuteAsync(IProgress<string> progress, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Start executing processing job {_inputData.ProcessingJobSequenceId}.");
+
+            try
+            {
+                // clear result at first
+                _result = new FhirToDataLakeProcessingJobResult();
+
+                progress.Report(JsonConvert.SerializeObject(_result));
+
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    _logger.LogInformation("Job is cancelled.");
+                    throw new OperationCanceledException();
+                }
+
+                // clean resources before process start
+                await CleanResourceAsync(cancellationToken);
+
+                _cacheResult = new CacheResult();
+                _outputFileIndexMap = new Dictionary<string, int>();
+
+                _typeFilters = _typeFilterParser.CreateTypeFilters(
+                    _filterConfiguration.FilterScope,
+                    _filterConfiguration.RequiredTypes,
+                    _filterConfiguration.TypeFilters).ToList();
+
+                switch (_filterConfiguration.FilterScope)
+                {
```

Comment:
Looks like 2 job type are almost different, can we split to 2 job type?

Reply:
these two job types are different during configuration, the core execution `ProcessFiltersAsync` are the same.

Reply:
Extract sub-functions for group/system scope type.